src/python/pants/task/task.py

```diff
@@ -238,6 +233,7 @@ def fingerprint(self):
     A task's fingerprint is only valid afer the task has been fully initialized.
     """
     hasher = sha1()
+    hasher.update(self.stable_name())
     hasher.update(self._options_fingerprint(self.options_scope))
```

Comment:
So, while `self._options_fingerprint` includes the options values, I don't know if it includes their fully-qualified names (ie, scoped)? Can you confirm that? If it does, then great.

If it doesn't, then a simple solution would be to do something here like:
```
hasher.update(self.options_scope)
```

Reply:
Thanks a lot for catching this! Fixed in 0b5694e69.

Reply:
I actually moved this into `self._options_fingerprint()` in 4d8c9c51b.